### PR TITLE
Feature/spire model fixes

### PR DESCRIPTION
**Pull request by abbas123456**
This will add notnull constraints, change dates into datetimes and fix a couple of errors from #96

### CODE REVIEW

File: tests/fixtures/factories.py [hunk 1]
```diff
@@ -46,19 +46,6 @@ class Meta:
...
-    withheld_status = factory.Faker('word')
-    batch_id = None
-    ela_id = factory.Faker('random_int', min=1, max=50)
-
-    class Meta:
-        model = SPIREApplication
-
-
 class SPIREBatchFactory(BaseFactory):
     start_date = factory.Faker('date_between', start_date='-2y', end_date='-1y')
```
Comment (marcuspp) [inline tests/fixtures/factories.py:50]: All faker for dates should be updated to date time functions https://faker.readthedocs.io/en/stable/providers/faker.providers.date_time.html#faker.providers.date_time.Provider.date_time_between_dates

File: app/db/models/external.py [hunk 2]
```diff
@@ -516,20 +525,20 @@ class SPIRERefDoNotReportValue(BaseModel):
     dnr_value = _col(_text, primary_key=True)
 
 
-class SPIREReturns(BaseModel):
-    __tablename__ = 'returns'
+class SPIREReturn(BaseModel):
+    __tablename__ = 'return'
```
Comment (marcuspp) [inline app/db/models/external.py:529]: Why is the table being renamed? I thought the table in oracle is `returns`?

**Comment by marcuspp** [comment]
Looks good just a few small comments :+1:

File: app/db/models/external.py (hunk 2 above)
Comment (abbas123456) [reply app/db/models/external.py:529]: I used the singular term for all table names and model names but this was missed out

**Pull request merged**